src/Elems/itemList.js

```diff
@@ -0,0 +1,210 @@
+import React from 'react';
+import Item from './item';
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import axios from 'axios';
+ 
+toast.configure()
+
+class ItemList extends React.Component {
+    state = {
+        filterStatus:'all',
+        markAll:true
+    }
+
+    delTodos = async (id, _id) => {
+        const itemIndex = this.props.data.findIndex(el => el.id === id);
+        try {
+            const res = await axios.delete('http://localhost:1234/products/'+_id+'/delete');
+            if (res.status == 200){
+                this.props.data.splice(itemIndex, 1); 
+                toast.error('You delete todo');               
+            }; 
+        }
+        catch (err) {
+        };
+        this.setState({data:this.props.data});
+    };
+    
+    onUndoneListBtnClick = () => {
+        this.setState({filterStatus:'undone'})
+    };
+
+    onDoneListBtnClick = () => {
+        this.setState({filterStatus:'done'})
+    };
+    
+    onAllListBtnClick = () => {
+        this.setState({filterStatus:'all'})
+    };
+
+    onMarkAllBtnClick = () => {
+        if (this.state.markAll == true){
+           this.onMarkAllDoneBtnClick()
+        }
+        else this.onMarkAllUndoneBtnClick();
+    };
+
+    onMarkAllDoneBtnClick = async () => {
+        try {
+            const res = await axios.put('http://localhost:1234/products/change',{
+                status:true
+            });
+                if (res.status == 200){
+                    return this.props.data.map((item) => {
+                    return item.status = true;
+                    }),
+                    this.setState({data:this.props.data, markAll:false}),
+                    toast.info('You marcked all as "done"');
+                };
+            }
+            catch (err) {
+            };
+    };
+
+    onMarkAllUndoneBtnClick = async () => {
+        try {
+            const res = await axios.put('http://localhost:1234/products/change',{
+                status:false
+            })
+                if (res.status == 200){
+                    return  this.props.data.map((item) => {
+                                return item.status = false
+                            }),
+                    this.setState({data:this.props.data, markAll:true}),
+                    toast.info('You marcked all as "undone"');
+                };
+        }
+        catch (err) {
+        };
+    }
+    
```

Comment:
The functions `onMarkAllDoneBtnClick` and `onMarkAllUndoneBtnClick` can be written as one function